File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() > t.length())
            return false;
        unordered_map<char, int> mp;
        for(auto i=0; i<s.length(); i++){
            mp[s[i]]++;
        }
        for(auto i=0; i<t.length(); i++){
            if(mp[t[i]] < 1)
                return false;
            else
                mp[t[i]]--;
        }
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length())
            return false;
        HashMap<Character, Integer> mp = new HashMap<>();
        for(char c : s.toCharArray()){
            mp.put(c, mp.getOrDefault(c, 0) + 1);
        }
        for(char c : t.toCharArray()){
            mp.put(c, mp.getOrDefault(c, 0) - 1);
        }
        for(char i : mp.keySet()){
            if(mp.get(i) != 0)
                return false;
        }
        return true;
    }
}